src/serlo-editor/core/sub-document/editor.tsx

```diff
@@ -74,6 +76,23 @@ export function SubDocumentEditor({ id, pluginProps }: SubDocumentProps) {
     [focused, id, dispatch, document]
   )
 
+  const noVisualFocusHandling = document?.plugin
+    ? document.plugin.startsWith('type-') ||
+      [EditorPluginType.Article, EditorPluginType.Rows].includes(
+        document.plugin as EditorPluginType
+      )
+    : true
+
+  const handleDomFocus = useCallback(() => {
+    const target = containerRef.current
+    if (!target) return
+    setTimeout(() => {
```

Comment:
>Activating slate sometimes takes two clicks or looses cursor position.

Removing this `setTimeout` wrapper resolves this bug. What problem does this `setTimeout` solve? Maybe we can think of another solution to that problem?

Reply:
Just saw you changed it, nice 👍